### PR TITLE
Fix and improve scheduling workflow

### LINKED ISSUE #16: Fix and improve scheduling workflow
- [x] Remove schedule when the unpublish date is cleared, https://github.com/humanmade/Unpublish/commit/99904012d5519ce08b1f4e58e1d1bb71613e509e.
- [x] Add/remove schedule by hooking into `{added,updated,deleted}_post_meta`, https://github.com/humanmade/Unpublish/commit/489fb7eeb69d078ab1ad258e5e711a3bff46a8cd.
- [x] Remove schedule when the post is trashed, https://github.com/humanmade/Unpublish/commit/5f2fdff9e7144d4c85e16a7be55521100abdc0e2.
- [x] Add schedule back when the post is untrashed (if the unpublish date is set and is later than current time), https://github.com/humanmade/Unpublish/commit/8f203488f00b37dd4b45f21ea1317baa5eb0b41c.

### PR DESCRIPTION
**Pull request by kucrut**
Fixes #16

### CODE REVIEW

**Comment by kucrut** [comment]
Closing this and opening a new one. Should've created the PR against `development`.

**Comment by kucrut** [comment]
Oh well, there's no `development` branch. Reopening 😄

**Comment by kucrut** [comment]
@peterwilsoncc There's one step missing from the last update; deleting the schedule when the unpublish date is cleared. This PR fixes it along with other improvements to the workflow. Please review.

File: unpublish.php [hunk 1]
```diff
@@ -123,7 +124,7 @@ public function render_unpublish_ui() {
 		$unpublish_timestamp = $this->get_unpublish_timestamp( get_the_ID() );
 		if ( ! empty( $unpublish_timestamp ) ) {
 			$local_timestamp = strtotime( get_date_from_gmt( date( 'Y-m-d H:i:s', $unpublish_timestamp ) ) );
-			$datetime_format = sprintf( __( '%s @ %s', 'unpublish' ), $this->date_format, $this->time_format );
+			$datetime_format = __( 'M j, Y @ H:i', 'unpublish' );
```
Comment (peterwilsoncc) [inline unpublish.php]: Translators note would be handy here, again borrowing from the core code. 
`/* translators: Unpublish box date format, see https://secure.php.net/date */`

**Comment by peterwilsoncc** [comment]
Added a note about a translator comment  and then happy if you merge it. 

Pete

**Comment by kucrut** [comment]
Thanks for the review @peterwilsoncc!

**Pull request merged**